src/qpx_mmc_css.cpp

```diff
@@ -0,0 +1,1909 @@
+      i_t3 = (i_t3 << 8) | i_t6;
+      i_t6 = p_css_tab4[i_t6];
+      i_t5 += i_t6 + i_t4;
+      if ((i_t5 & 0xff) != p_buffer[i]) {
+        break;
+      }
+
+      i_t5 >>= 8;
+    }
+
+    if (i == 10) {
+      /* Do 4 backwards steps of iterating t3 to deduce initial state */
+      i_t3 = i_candidate;
+      for (i = 0; i < 4; i++) {
+        i_t1 = i_t3 & 0xff;
+        i_t3 = (i_t3 >> 8);
+        /* easy to code, and fast enough bruteforce
+         * search for byte shifted in */
+        for (j = 0; j < 256; j++) {
+          i_t3 = (i_t3 & 0x1ffff) | (j << 17);
+          i_t6 = (((((((i_t3 >> 3) ^ i_t3) >> 1) ^ i_t3) >> 8) ^ i_t3) >> 5) &
+                 0xff;
+          if (i_t6 == i_t1) {
+            break;
+          }
+        }
+      }
+
+      i_t4 = (i_t3 >> 1) - 4;
+      for (i_t5 = 0; i_t5 < 8; i_t5++) {
+        if (((i_t4 + i_t5) * 2 + 8 - ((i_t4 + i_t5) & 7)) == i_t3) {
+          p_key[0] = i_try >> 8;
+          p_key[1] = i_try & 0xFF;
+          p_key[2] = ((i_t4 + i_t5) >> 0) & 0xFF;
+          p_key[3] = ((i_t4 + i_t5) >> 8) & 0xFF;
+          p_key[4] = ((i_t4 + i_t5) >> 16) & 0xFF;
+          i_exit = i_try + 1;
+        }
+      }
+    }
+  }
+
+  if (i_exit >= 0) {
+    p_key[0] ^= p_sector_seed[0];
+    p_key[1] ^= p_sector_seed[1];
+    p_key[2] ^= p_sector_seed[2];
+    p_key[3] ^= p_sector_seed[3];
+    p_key[4] ^= p_sector_seed[4];
+  }
+
+  return i_exit;
+}
+
+/*****************************************************************************
+ * DecryptTitleKey
+ *****************************************************************************
+ * Decrypt the title key using the disc key.
+ * p_disc_key: result, the 5 byte disc key
+ * p_titlekey: the encrypted title key, gets overwritten by the decrypted key
+ *****************************************************************************/
+static void css_DecryptTitleKey(dvd_key_t p_disc_key, dvd_key_t p_titlekey) {
+  css_DecryptKey(0xff, p_disc_key, p_titlekey, p_titlekey);
+}
+
+/*****************************************************************************
+ * CrackTitleKey: try to crack title key from the contents of a VOB.
+ *****************************************************************************
+ * This function is called by _dvdcss_titlekey to find a title key, if we've
+ * chosen to crack title key instead of decrypting it with the disc key.
+ * The DVD should have been opened and be in an authenticated state.
+ * i_pos is the starting sector, i_len is the maximum number of sectors to read
+ *****************************************************************************/
+static int css_CrackTitleKey(drive_info *drive, int i_pos, int i_len,
+                             dvd_key_t p_titlekey) {
+  unsigned char p_buf[DVDCSS_BLOCK_SIZE];
+  const unsigned char p_packstart[4] = {0x00, 0x00, 0x01, 0xba};
+  int i_reads = 0;
+  int i_encrypted = 0;
+  int b_stop_scanning = 0;
+  int b_read_error = 0;
+  int i_ret;
```

Comment:
Codacy found an issue: [The scope of the variable 'i_ret' can be reduced.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)